di_website/publications/models.py

```diff
@@ -1,57 +1,56 @@
-from num2words import num2words
-from itertools import chain
 import operator
 from functools import reduce
+from itertools import chain
 
 from django import forms
-from django.db import models
-from django.db.models import Q, FloatField, Value
-from django.core.paginator import EmptyPage, PageNotAnInteger, Paginator
 from django.contrib.contenttypes.models import ContentType
+from django.core.paginator import EmptyPage, PageNotAnInteger, Paginator
+from django.db import models
+from django.db.models import FloatField, Q, Value
 from django.utils.functional import cached_property
 from django.utils.text import slugify
-
 from modelcluster.contrib.taggit import ClusterTaggableManager
 from modelcluster.fields import ParentalKey
 from modelcluster.models import ClusterableModel
-
-from wagtail.core import hooks
-from wagtail.core.models import Page, Orderable
-from wagtail.core.fields import RichTextField, StreamField
-from wagtail.core.blocks import (
-    CharBlock,
-    PageChooserBlock,
-    StructBlock,
-    URLBlock
-)
-from wagtail.admin.edit_handlers import FieldPanel, InlinePanel, StreamFieldPanel, PageChooserPanel
-from wagtail.snippets.models import register_snippet
-from wagtail.snippets.edit_handlers import SnippetChooserPanel
+from num2words import num2words
+from taggit.models import Tag, TaggedItemBase
```

Comment:
Move these to the top grouping ... In case you've not realised, we group our imports by module.. first grouping is usually for random ones like these.

Reply:
Make sure u add a blank line between these groupings ... I see you've removed some of them... I see the following groupings...  `random`, `django`, `modelcluster`, `wagtail`, `di_website`

Reply:
Grouped imports and added new line separators